File: CSharp/HelloUniverse.Tests/LinkedLists/MergeKSortedListsTests.cs
using HelloUniverse.LinkedLists;
using NUnit.Framework;
using System.Collections.Generic;

namespace HelloUniverse.Tests.LinkedLists
{
    [TestFixture]
    public class MergeKSortedListsTests
    {
        public delegate MergeKSortedLists.ListNode MergeDelegate(MergeKSortedLists.ListNode[] lists);

        [TestCaseSource(nameof(TestCases))]
        public void TestCopy(MergeDelegate merge)
        {
            Assert.IsNull(merge(null));
            Assert.IsNull(merge(new MergeKSortedLists.ListNode[] { }));

            var l1 = new MergeKSortedLists.ListNode(1)
            {
                next = new MergeKSortedLists.ListNode(2)
                {
                    next = new MergeKSortedLists.ListNode(4)
                }
            };

            var l2 = new MergeKSortedLists.ListNode(3)
            {
                next = new MergeKSortedLists.ListNode(4)
                {
                    next = new MergeKSortedLists.ListNode(5)
                }
            };

            var l3 = new MergeKSortedLists.ListNode(2)
            {
                next = new MergeKSortedLists.ListNode(3)
                {
                    next = new MergeKSortedLists.ListNode(6)
                }
            };

            Assert.AreEqual("1->2->2->3->3->4->4->5->6", merge(new MergeKSortedLists.ListNode[] { l1, l2, l3 }).ToString());

        }


        private static IEnumerable<TestCaseData> TestCases
        {
            get
            {
                yield return
                    new TestCaseData(new MergeDelegate(MergeKSortedLists.Merge)).SetName(
                        nameof(MergeKSortedLists.Merge));
            }
        }
    }
}

File: Java/src/unittests/MergeKSortedListsTests.java
package unittests;

import static org.junit.Assert.*;

import org.junit.Test;

import linkedlists.MergeKSortedLists;

public class MergeKSortedListsTests {

	@Test
	public void testMerge() {
		assertNull(MergeKSortedLists.merge(null));
		assertNull(MergeKSortedLists.merge(new MergeKSortedLists.ListNode[] { }));

		MergeKSortedLists.ListNode l1 = new MergeKSortedLists.ListNode(1);
		l1.next =new MergeKSortedLists.ListNode(2);
		l1.next.next =new MergeKSortedLists.ListNode(4);
		
		MergeKSortedLists.ListNode l2 = new MergeKSortedLists.ListNode(3);
		l2.next =new MergeKSortedLists.ListNode(4);
		l2.next.next =new MergeKSortedLists.ListNode(5);
		
		MergeKSortedLists.ListNode l3 = new MergeKSortedLists.ListNode(2);
		l3.next =new MergeKSortedLists.ListNode(3);
		l3.next.next =new MergeKSortedLists.ListNode(6);

        

       assertEquals("1->2->2->3->3->4->4->5->6->", MergeKSortedLists.merge(new MergeKSortedLists.ListNode[] { l1, l2, l3 }).toString());
	}

}
